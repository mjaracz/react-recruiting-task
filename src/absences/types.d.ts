import { Product } from './redux/types'

export interface ItemProps {
  deleteItem: (id: number) => void
  absences: AbsencePerson
}

export interface AbsencesState {
  isLoading: boolean
  list: AbsencePerson[]
  error: Error
}

export interface AbsencePerson {
  admitterId: number | null
  admitterNote: string
  confirmedAt: string
  createAt: string
  crewId: number
  endDate: string
  id: number
  memberNote: string
  rejectedAt: number | null
  startDate: string
  type: string
  userId: number
}