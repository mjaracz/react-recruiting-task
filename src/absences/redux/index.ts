import {absencesReducer} from './reducer'
import {absencesWatcher} from './saga'

export {absencesReducer, absencesWatcher}