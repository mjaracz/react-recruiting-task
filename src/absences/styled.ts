import { makeStyles } from '@material-ui/styles'

export const useStyle = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
    flexWrap: 'wrap',
    overflowY: 'auto',
    marginTop: 120,
    '& .MuiList-dense': {
      marginTop: 80,
      paddingButton: '0 !important'
    },
  },
  item: {
    backgroundColor: '#f6f6f6',
    margin: 20,
    borderRadius: 15,
    width: 350,
  },
  itemText: {
    '& .MuiListItemText-secondary': {
      color: '#f5005780',
    },
  },
  itemDescText: {
    margin: 8,
  },
  itemDeleteAction: {
    right: 40,
  },
})
