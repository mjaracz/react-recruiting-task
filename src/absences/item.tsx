import React, { FC } from 'react'
import { ItemProps } from './types'
import { useStyle } from './styled'
import {
  IconButton,
  ListItem,
  ListItemAvatar,
  ListItemSecondaryAction,
  ListItemText,
} from '@material-ui/core'
import { Deck } from '@material-ui/icons'
import DeleteIcon from '@material-ui/icons/Delete'

export const AbsenceItem: FC<ItemProps> = ({
  absences,
}) => {
  const { item, itemText, itemDescText, itemDeleteAction } = useStyle()
  return (
    <ListItem className={item}>
      <ListItemAvatar>
        <Deck />
      </ListItemAvatar>
      <ListItemText
        className={itemText}
        primary={absences.type}
        secondary={`From ${absences.startDate} to ${absences.endDate}`}
      />
      <ListItemText
        className={itemDescText}
        primary={`User Id ${absences.userId}`}
      />
      <ListItemSecondaryAction className={itemDeleteAction}>
        <IconButton edge="end" aria-label="delete-product" >
          <DeleteIcon />
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  )
}
