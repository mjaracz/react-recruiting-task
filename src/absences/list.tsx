import React, {FC} from 'react'
import {useStyle} from './styled'
import {useList} from './hooks/useList'
import {CircularProgress, List} from '@material-ui/core'
import {AbsenceItem} from './item'

export const AbsencesList: FC = () => {
  const {absences} = useList()
  const {root} = useStyle()

  return (
    <div className={root}>
      {absences.isLoading
        ? <CircularProgress size={80}/>
        : (
          <List>
            {absences.list.map((item, index) => <AbsenceItem key={index} absences={item} />)}
          </List>
        )}
    </div>
  )
}