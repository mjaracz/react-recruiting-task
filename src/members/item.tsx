import React, {FC} from 'react'
import {IconButton, ListItem, ListItemAvatar, ListItemSecondaryAction, ListItemText} from '@material-ui/core'
import {Favorite} from '@material-ui/icons'
import DeleteIcon from '@material-ui/icons/Delete'
import {useStyle} from '../absences/styled'
import {Member} from './types'

export const MemberItem: FC<{member: Member}> = ({
  member
}) => {
  const {item, itemText, itemDeleteAction} = useStyle()
  return (
    <ListItem className={item}>
      <ListItemAvatar>
        <Favorite/>
      </ListItemAvatar>
      <ListItemText
        className={itemText}
        primary={member.name}
        secondary={`User Id ${member.userId}`}
      />
      <ListItemSecondaryAction className={itemDeleteAction}>
        <IconButton edge="end" aria-label="delete-product" >
          <DeleteIcon />
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  )
}
