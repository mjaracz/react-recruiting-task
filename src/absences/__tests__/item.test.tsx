import React from 'react'
import {render} from '@testing-library/react'
import {AbsenceItem} from '../item'
import * as absences from '../../api/json_files/absences.json'

describe('<AbsenceItem/>', () => {
  it('should be defined and match snapshot', function () {
    const {baseElement} = render(<AbsenceItem absences={absences.payload[3]} />)
    expect(baseElement).toMatchSnapshot()
  });
})