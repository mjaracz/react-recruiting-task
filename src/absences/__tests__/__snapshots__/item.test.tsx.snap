// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AbsenceItem/> should be defined and match snapshot 1`] = `
<body>
  <div>
    <li
      class="MuiListItem-container"
    >
      <div
        class="MuiListItem-root makeStyles-item-2 MuiListItem-gutters MuiListItem-secondaryAction"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 9L12 2 2 9h9v13h2V9z"
            />
            <path
              d="M4.14 12l-1.96.37.82 4.37V22h2l.02-4H7v4h2v-6H4.9zM19.1 16H15v6h2v-4h1.98l.02 4h2v-5.26l.82-4.37-1.96-.37z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root makeStyles-itemText-3 MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            vacation
          </span>
          <p
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            From 2017-02-02 to 2017-02-06
          </p>
        </div>
        <div
          class="MuiListItemText-root makeStyles-itemDescText-4"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            User Id 5293
          </span>
        </div>
      </div>
      <div
        class="MuiListItemSecondaryAction-root makeStyles-itemDeleteAction-5"
      >
        <button
          aria-label="delete-product"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
  </div>
</body>
`;
